### PR TITLE
Geometry library extension for pose and point

### PR DESCRIPTION
**Pull request by jveijck**

### CODE REVIEW

File: libraries/geometry/point.cpp [hunk 1]
```diff
@@ -90,3 +92,43 @@ Point& Point::operator/=(double f)
...
+}
+
+double Point::angle( const Point& point) const {
+    return atan2(this->y - point.y, this->x - point.x);
+}
+
+/**
+ * Normalizes the point such that its norm is 1.0 and direction unchanged.
+ *
+ * @return Normalized point. When input point has length 0, the result is (1,0)
```
Comment (janfeitsma) [inline libraries/geometry/point.cpp:121]: That is not obvious and might cause trouble. 

Why not throw a runtime error "division-by-zero" instead?

File: libraries/geometry/point.hpp [hunk 2]
```diff
@@ -20,7 +23,7 @@ class Point
 
     // basic operations
     void reset();
-    double size() const; // sqrt(x^2+y^2+z^2)
```
Comment (janfeitsma) [inline libraries/geometry/point.hpp:23]: Use cases for point also include ball, which typically has z > 0. So I propose to make use of z and not drop it.

**Review by janfeitsma — CHANGES_REQUESTED** [review]

File: libraries/geometry/point.hpp (hunk 2 above)
Comment (jveijck) [reply libraries/geometry/point.hpp:23]: the class point does not have no z. Therefore it should not be in the comment

File: libraries/geometry/point.cpp (hunk 1 above)
Comment (jveijck) [reply libraries/geometry/point.cpp:121]: agree, will throw std::range_error (range errors in internal computations)

File: libraries/geometry/point.hpp (hunk 2 above)
Comment (janfeitsma) [reply libraries/geometry/point.hpp:23]: Agree

**Review by janfeitsma — APPROVED** [review]

**Pull request merged**